src/command.ts

```diff
@@ -200,49 +204,120 @@ export abstract class IronfishCommand extends Command {
+      internalOverrides: internalOverrides,
+      configName: typeof configFlag === 'string' ? configFlag : undefined,
+      dataDir: typeof dataDirFlag === 'string' ? dataDirFlag : undefined,
+      logger: this.logger,
+    })
+
+    this.walletConfig = new WalletConfig(this.sdk.fileSystem, this.sdk.dataDir)
+    await this.walletConfig.load()
+
+    const walletConfigOverrides: Partial<WalletConfigOptions> = {}
+
+    // TODO: This is here for backawrds compatability to migrate wallet configs in the old
+    this.migrateOldConfig(
+      this.sdk.config,
+      this.walletConfig,
+      walletConfigOverrides,
+    )
+
     const walletNodeUseIpcFlag = getFlag(flags, WalletNodeUseIpcFlagKey)
     if (typeof walletNodeUseIpcFlag === 'boolean') {
-      configOverrides.walletNodeIpcEnabled = walletNodeUseIpcFlag
+      walletConfigOverrides.walletNodeIpcEnabled = walletNodeUseIpcFlag
     }
 
     const walletNodeIpcPathFlag = getFlag(flags, WalletNodeIpcPathFlagKey)
     if (typeof walletNodeIpcPathFlag === 'string') {
-      configOverrides.walletNodeIpcPath = walletNodeIpcPathFlag
+      walletConfigOverrides.walletNodeIpcPath = walletNodeIpcPathFlag
     }
 
     const walletNodeUseTcpFlag = getFlag(flags, WalletNodeUseTcpFlagKey)
     if (typeof walletNodeUseTcpFlag === 'boolean') {
-      configOverrides.walletNodeTcpEnabled = walletNodeUseTcpFlag
+      walletConfigOverrides.walletNodeTcpEnabled = walletNodeUseTcpFlag
     }
 
     const walletNodeTcpHostFlag = getFlag(flags, WalletNodeTcpHostFlagKey)
     if (typeof walletNodeTcpHostFlag === 'string') {
-      configOverrides.walletNodeTcpHost = walletNodeTcpHostFlag
+      walletConfigOverrides.walletNodeTcpHost = walletNodeTcpHostFlag
     }
 
     const walletNodeTcpPortFlag = getFlag(flags, WalletNodeTcpPortFlagKey)
     if (typeof walletNodeTcpPortFlag === 'number') {
-      configOverrides.walletNodeTcpPort = walletNodeTcpPortFlag
+      walletConfigOverrides.walletNodeTcpPort = walletNodeTcpPortFlag
     }
 
     const walletNodeTcpTlsFlag = getFlag(flags, WalletNodeTcpTlsFlagKey)
     if (typeof walletNodeTcpTlsFlag === 'boolean') {
-      configOverrides.walletNodeTlsEnabled = walletNodeTcpTlsFlag
+      walletConfigOverrides.walletNodeTlsEnabled = walletNodeTcpTlsFlag
     }
 
     const walletNodeAuthFlag = getFlag(flags, WalletNodeAuthFlagKey)
     if (typeof walletNodeAuthFlag === 'string') {
-      configOverrides.walletNodeRpcAuthToken = walletNodeAuthFlag
+      walletConfigOverrides.walletNodeRpcAuthToken = walletNodeAuthFlag
     }
 
-    this.sdk = await IronfishSdk.init({
-      pkg: IronfishCliPKG,
-      configOverrides: configOverrides,
-      internalOverrides: internalOverrides,
-      configName: typeof configFlag === 'string' ? configFlag : undefined,
-      dataDir: typeof dataDirFlag === 'string' ? dataDirFlag : undefined,
-      logger: this.logger,
-    })
+    Object.assign(this.walletConfig.overrides, walletConfigOverrides)
+  }
+
+  migrateOldConfig(
```

Comment:
I think we need to save the migrated values to the walletConfig file

Reply:
+1 to this comment if we're creating a new `config.wallet.json` file

Reply:
hmm, I can buy that. It makes sense.

Reply:
Fixed